src/color.cc

```diff
@@ -15,23 +17,249 @@
 #define WHITESPACE \
   while (' ' == *str) ++str;
 
+#define WHITESPACE_OR_COMMA \
+  while (' ' == *str || ',' == *str) ++str;
+
+
+/*
+ * Parse integer value
+ */
+
+template <typename parsed_t>
+static bool parseInteger(const char** pStr, parsed_t *pParsed)
```

Comment:
why are we manually doing this integer parsing? I'm looking for a good reason in here but it's escaping me so far.